### ISSUE TITLE
Settings for Wacom Stylus Buttons are not properly respected

### ISSUE THREAD

**Issue by ysooqe** [xournalpp/xournalpp#1181]
**Affects versions :**
 - OS: Linux
 - Version of Xournal++ 1.0.12

**Describe the bug**
I set one of the buttons on the Wacom Stylus to the eraser (since the Stylus has no default eraser) with the options Thin and Standard. Now, when using this button to erase something, it actually just uses the thickness of the eraser tool that was last set. So if I switch from pen to eraser, set the thickness to the largest one, switch back to pen, draw something and then want to erase with the stylus button, it always is the thickness of the eraser tool instead of the preset thickness "Thin" that I set in the settings.
For the type of the stylus button eraser, it somehow works almost the other way around: The eraser tool will always be set to the same setting (Standard, Whiteout, Deleting Stroke) that the stylus button eraser tool is set to after using the stylus button eraser. So if I set the stylus button eraser type to Delete Stroke, then draw something with the pen tool, then go to the eraser tool, select Standard, using the tool I can just erase whatever is under the cursor. Now I use the stylus button eraser instead, this time it deletes strokes (as expected because thats the behaviour that was set in the settings) but now, when switching to the eraser tool, this one is set to Delete Stroke aswell.
To summarise: The setting of the stylus button as an eraser and the eraser tool do not act independet of each other but interact with each others settings.

**To Reproduce**
as given in the description.

**Expected behavior**
settings for stylus button eraser and erasor tool are independet of each other

**Input Logs**
[debug_gdk_input.log](https://github.com/xournalpp/xournalpp/files/3227039/debug_gdk_input.log)
[debug_input.log](https://github.com/xournalpp/xournalpp/files/3227040/debug_input.log)

**Comment by LittleHuba** [comment]
Are you using the old or the new input system for this?

**Comment by LittleHuba** [comment]
And please provide clear steps to reproduce this. Your description is not really clear enough.

1. Set settings x & y
2. Press button z
3. ...

**Comment by ysooqe** [comment]
It happens with both the new and the old input system.

And a clearer explanation:
Eraser tool: Eraser from the toolbar.
Stylus Button Eraser: Eraser that is set to a button of the stylus.

Steps:
1. Set one of the Stylus Buttons to eraser (in the settings), with additional opions "Thin" and "Standard".
2. draw something (e.g. a simple line)
3. switch from pen to the eraser tool, chose the thickest setting
4. erase the line using the eraser tool
5. switch back to pen and draw something else
6. now press the stylus button that is set to the eraser.

expected behavior: the eraser of the stylus button should be "Thin", as specified in the settings.

actual behavior: it is set to the thickest setting and as it turns out, it always uses the current thickness setting of the eraser tool from the toolbar.

**Comment by ysooqe** [comment]
I cloned the repo today and built the latest nigthly (16/07) and the problem just reversed.
So now, the eraser tool from the toolbar will always switch to whatever the Button setting is (so basically the problem is reversed now).
Maybe I do not understand the features correctly, but in Xournal it was just like this:
Button-tools and toolbar-tools where two separate things and their respective settings were not intefering with each other, meaning: When setting the Eraser on the Button to "thin" and the eraser from the toolbar to "thick", these setting would stay that way when the other was used.
Because this enables a very convenient workflow: pressing the button on the pen to erase a fine line (setting "thin") and switching over to the eraser tool to when theres a need to erase a larger area (setting "thick").

**Comment by Technius** [comment]
This bug has been fixed in master. Probably by #2521.
